dict.go

```diff
@@ -25,10 +26,15 @@ import (
 
 // Dictionary is the zap representation of the term dictionary
 type Dictionary struct {
-	sb        *SegmentBase
-	field     string
-	fieldID   uint16
-	fst       *vellum.FST
+	sb      *SegmentBase
+	field   string
+	fieldID uint16
+	fst     *vellum.FST
+
+	// if the dictionary is shared across multiple threads
+	// we need to protect the fstReader with a mutex
+	// since it is not thread safe
+	m         sync.Mutex
```

Comment:
Sure the reader is not thread safe - why is guarding access here with a mutex a concern now?

Reply:
the dictionary object is always freshly created for any get Dictionary(field) call by a thread. The dictionary object will not be shared across any thread and will remain unique to each thread. With the new approach suggested here, the dictionary for the "_id" field will be created only once and cached in the segment for its lifetime. So it is subject to being shared across threads due to the cache being a shared resource. Hence, here it becomes necessary to guard the fst reader with a mutex.